lib/OpenQA/WebAPI/Controller/Main.pm

```diff
@@ -177,15 +177,81 @@ sub add_comment {
     my $group = $self->app->schema->resultset("JobGroups")->find($self->param('groupid'));
     return $self->reply->not_found unless $group;
 
+    # only logged in users can add comments
+    if (!$self->current_user) {
+        $self->flash('info', 'The comment couldn\'t be added because you\'re not logged in anymore');
+        return $self->redirect_to('group_overview');
+    }
+
     my $rs = $group->comments->create(
         {
             text    => $self->param('text'),
             user_id => $self->current_user->id,
         });
+
     $self->emit_event('openqa_user_comment', {id => $rs->id});
     $self->flash('info', 'Comment added');
     return $self->redirect_to('group_overview');
 }
 
+sub edit_comment {
+    my ($self) = @_;
+
+    $self->validation->required('text');
+    $self->validation->required('comment_id');
+    my $comment_id = int($self->param("comment_id"));
+
+    my $group = $self->app->schema->resultset("JobGroups")->find($self->param('groupid'));
+    return $self->reply->not_found unless $group;
+
+    # only logged in users can edit comments
+    if (!$self->current_user) {
+        $self->flash('info', 'The comment couldn\'t be edited because you\'re not logged in anymore');
+        return $self->redirect_to('group_overview');
+    }
```

Comment:
you have the same code twice now. See here: http://clean-code-developer.de/die-grade/roter-grad/#Dont_Repeat_Yourself_DRY